src/lib/support/jsontlv/JsonToTlv.cpp

```diff
@@ -0,0 +1,418 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <algorithm>
+#include <json/json.h>
+#include <lib/support/Base64.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/jsontlv/ElementTypes.h>
+#include <lib/support/jsontlv/JsonToTlv.h>
+
+namespace chip {
+
+namespace {
+
+std::vector<std::string> splitIntoFieldsBySeparator(const std::string & input, char separator)
+{
+    std::vector<std::string> substrings;
+    std::stringstream ss(input);
+    std::string substring;
+
+    while (std::getline(ss, substring, separator))
+    {
+        substrings.push_back(substring);
+    }
+
+    return substrings;
+}
+
+struct ElementTypeContext
+{
+    TLV::TLVType tlvType = TLV::kTLVType_NotSpecified;
+    bool isDouble        = false;
+};
+
+CHIP_ERROR JsonTypeStrToTlvType(const char * elementType, ElementTypeContext & type)
+{
+    if (strcmp(elementType, kElementTypeInt) == 0)
+    {
+        type.tlvType = TLV::kTLVType_SignedInteger;
+    }
+    else if (strcmp(elementType, kElementTypeUInt) == 0)
+    {
+        type.tlvType = TLV::kTLVType_UnsignedInteger;
+    }
+    else if (strcmp(elementType, kElementTypeBool) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Boolean;
+    }
+    else if (strcmp(elementType, kElementTypeFloat) == 0)
+    {
+        type.tlvType  = TLV::kTLVType_FloatingPointNumber;
+        type.isDouble = false;
+    }
+    else if (strcmp(elementType, kElementTypeDouble) == 0)
+    {
+        type.tlvType  = TLV::kTLVType_FloatingPointNumber;
+        type.isDouble = true;
+    }
+    else if (strcmp(elementType, kElementTypeBytes) == 0)
+    {
+        type.tlvType = TLV::kTLVType_ByteString;
+    }
+    else if (strcmp(elementType, kElementTypeString) == 0)
+    {
+        type.tlvType = TLV::kTLVType_UTF8String;
+    }
+    else if (strcmp(elementType, kElementTypeNull) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Null;
+    }
+    else if (strcmp(elementType, kElementTypeStruct) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Structure;
+    }
+    else if (strncmp(elementType, kElementTypeArray, strlen(kElementTypeArray)) == 0)
+    {
+        type.tlvType = TLV::kTLVType_Array;
+    }
+    else
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+bool isUnsignedInteger(const std::string & s)
```

Comment:
```suggestion
bool IsUnsignedInteger(const std::string & s)
```

And similar for the other helper functions, especially since some of them are already starting with an uppercase letter.

Reply:
This issue is still unresolved: https://github.com/project-chip/connectedhomeip/pull/27635#discussion_r1275432007

Stopping here for now; there's a bunch of issues here that are not the Common Profile problems that have not been addressed.

Reply:
updated